### PR TITLE
use PEP 589 syntax for TypeDict

### PR DESCRIPTION
**Pull request by holmanb**
```
Use PEP 589 syntax for TypeDict annotation.

Also fixes previously broken typing MetaSchema typing implementation.
```

### CODE REVIEW

File: cloudinit/importer.py [hunk 1]
```diff
@@ -12,21 +12,19 @@
...
-            "title": str,
-            "description": str,
-            "distros": typing.List[str],
-            "examples": typing.List[str],
-            "frequency": str,
-        },
-    )
+if sys.version_info >= (3, 8):
+
+    class MetaSchema:
```
Comment (TheRealFalcon) [inline cloudinit/importer.py:19]: This needs to inherit from TypedDict.

**Review by TheRealFalcon — COMMENTED** [review]
The old version had 
```python
if sys.version_info >= (3, 8) and hasattr(typing, "TypeDict"):
```
That's `TypeDict` rather than `TypedDict`, so this never actually worked.

In order for this proposed version to work, we also need to add this type to every `meta` definition because apparently you can't coerce a raw dict into a TypedDict. E.g., on `cc_apk_configure`, I currently see:
image

Can we update these meta definitions in this PR too?

**Comment by holmanb** [comment]
@TheRealFalcon I rebased on the new mypy commit and it looks like the fix you suggested allows us to remove a couple of files from the "exclude" list. This is ready for re-review.

**Review by TheRealFalcon — APPROVED** [review]
Nice! LGTM.

**Pull request merged**